tests/ut/backends/test_valkey.py

```diff
@@ -240,3 +240,37 @@ def test_build_key_double_dot(
 
     def test_build_key_no_namespace(self, valkey_cache):
         assert valkey_cache.build_key(Keys.KEY, namespace=None) == Keys.KEY
+
+    def test_custom_serializer(self, valkey_config):
+        assert isinstance(
+            ValkeyCache(config=valkey_config, serializer=PickleSerializer()).serializer,
+            PickleSerializer,
+        )
+
+    async def test_default_key_builder(self, valkey_config):
+        namespace = "namespace"
+        built_key = f"{namespace}:{Keys.KEY.value}"
+        async with ValkeyCache(
+            config=valkey_config,
+        ) as default:
+            await default.set(Keys.KEY.value, "value", namespace=namespace)
+
+        async with ValkeyCache(
+            config=valkey_config, key_builder=lambda k, ns: f"{ns}:{k}" if ns else k
+        ) as explicit:
+            assert await explicit.exists(Keys.KEY.value, namespace=namespace)
+            assert await explicit._exists(built_key)
```

Comment:
I think I'd personally bypass any of the abstraction for this assertion:
```suggestion
            await default.set(Keys.KEY.value, "value", namespace=namespace)
            assert await default.client.exists([built_key])
```

Reply:
I'm okay with that, but could you please elaborate why are do you want to tight the test to implementation?